service-worker.js

```diff
@@ -85,6 +85,17 @@ var regexpExcludedURLSchema = /^(?:file|chrome-extension|example-extension):/i;
  */
 const regexpZIMUrlWithNamespace = /(?:^|\/)([^/]+\/)([-ABCIJMUVWX])\/(.+)/;
 
+/**
+ * Pattern to parse the first offset of a "range" request header
+ * NB: this only reads the first offset of the first byte range, where the spec allows several ranges, and several units.
+ * See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range
+ * But, in our case, we send a header to tell the browser we only accept the bytes unit.
+ * I did not see multiple ranges asked by a browser.
+ * 
+ * @type {RegExp}
+ */
+const regexpByteRangeHeader = /^[ \t]*bytes=(\d+)-/;
```

Comment:
With `[ \t]`, are you attempting to match whitespace? If so, you could just use `\s` (not in square brackets. `\s` matches any whitespace, including spaces, tabs, newlines (which you won't have here of course). NB I haven't tested.

Reply:
You're right, I'll change.
I tend to have this habit because `\s` is not supported in all regular expression implementations. But it is in javascript so let's use it